core/src/trezor/wire/codec/codec_v1.py

```diff
@@ -25,7 +25,12 @@ async def read_message(iface: WireInterface, buffer: utils.BufferType) -> Messag
     read = loop.wait(iface.iface_num() | io.POLL_READ)
 
     # wait for initial report
-    report = await read
+    msg_len = await read
+    report = bytearray(msg_len)
+    read_len = iface.read(report, 0, msg_len)
+    if read_len != msg_len:
```

Comment:
these checks are kind of annoying... could we get rid of them somehow?
on usb we don't really need to know `msg_len` because it is fixed, so i'm guessing this is for BT with variable length packets?
perhaps `read()` should guarantee that it reads full `limit` (or buffer size) and exception out otherwise?
or, can we guarantee that if we got `msg_len` out of `await read`, the next read must return that many bytes? (i mean we could always do the full read into a buffer like we do on unix, i suppose...)

Reply:
how about this: https://github.com/trezor/trezor-firmware/commit/7d9a53c06946b51d4738f5e0c1b8d971f8807b8c

- the guarantee that msg_len is equal to real read data is implemented
- exception if buffer space is insufficient
- exception if for any reason the underlying USB implementation reads message of different length, but modification of the buffer is allowed under such circumstances
- limit is removed, assumed that you always read the whole message of the msg_len size. (or do we ever want to allow reading partials, even if it means losing the rest of the data?)